metadata-ingestion/src/datahub/ingestion/source/neo4j/neo4j_source.py

```diff
@@ -40,35 +33,35 @@
     StatefulIngestionReport,
     StatefulIngestionSourceBase,
 )
-from datahub.metadata.com.linkedin.pegasus2avro.schema import SchemaFieldDataType
-from datahub.metadata.schema_classes import (
-    AuditStampClass,
-    BooleanTypeClass,
-    DatasetPropertiesClass,
-    DateTypeClass,
-    NullTypeClass,
-    NumberTypeClass,
-    OtherSchemaClass,
-    SchemaFieldClass,
-    SchemaMetadataClass,
-    StringTypeClass,
-    SubTypesClass,
-    UnionTypeClass,
-)
+from datahub.sdk.dataset import Dataset
 
 log = logging.getLogger(__name__)
 logging.basicConfig(level=logging.INFO)
 
-_type_mapping: Dict[Union[Type, str], Type] = {
-    "list": UnionTypeClass,
-    "boolean": BooleanTypeClass,
-    "integer": NumberTypeClass,
-    "local_date_time": DateTypeClass,
-    "float": NumberTypeClass,
-    "string": StringTypeClass,
-    "date": DateTypeClass,
-    "node": StringTypeClass,
-    "relationship": StringTypeClass,
+# Neo4j object types
+_NODE = "node"
+_RELATIONSHIP = "relationship"
+
+# TODO: These DataHub schema field type constants should ideally be defined in SDK v2
```

Comment:
@hsheth2 did you consider adding an enum for these string data type values in SDK v2?

I found this is internally handled with:

https://github.com/datahub-project/datahub/blob/446f31f1bd69c0aeb7fa404fb586c3518b4c6564/metadata-ingestion/src/datahub/sdk/dataset.py#L798-L801

which is dynamic and depending on the platform:

https://github.com/datahub-project/datahub/blob/22e382d1beee273148da94261ddf9f6b435f3aaa/metadata-ingestion/src/datahub/ingestion/source/sql/sql_types.py#L484-L511

So the enum approach may be too static 🤔